src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,244 @@
+                    // Should not reach this case due to regex check above
+                    throw new InvalidFileFormatException();
+                }
+                loadedTasks.add(newTask);
+            }
+            this.tasks = loadedTasks;
+        } catch (InvalidFileFormatException e) {
+            Ui.printMessage(e.getMessage() + "\n");
+        }
+    }
+
+    /**
+     * Returns the formatted string for the "list" command.
+     *
+     * @return Formatted string of tasks.
+     */
+    public String list() {
+        StringBuilder str = new StringBuilder("Here are the tasks in your list:\n");
+        for (int i = 0; i < this.tasks.size(); i++) {
+            str.append(i + 1).append(".").append(this.tasks.get(i)).append("\n");
+        }
+
+        return str.toString();
+    }
+
+    /**
+     * Marks the task at the specified index as done.
+     *
+     * @param command The input command from the user.
+     * @return Formatted string containing the task marked as done, to be displayed to the user.
+     * @throws InvalidTaskIndexException When the index is out of bounds.
+     * @throws InvalidFormatException    When the input command is of the wrong format.
+     */
+    public String done(String command) throws InvalidTaskIndexException, InvalidFormatException {
+        int taskIdx = Parser.getIndexFromCommand(command);
+
+        // Handle invalid index
+        if (taskIdx >= 1 && taskIdx <= tasks.size()) {
+            Task t = tasks.get(taskIdx - 1);
+            t.markAsDone();
+            return String.format("Nice! I've marked this task as done:\n  %s\n", t);
+        } else {
+            throw new InvalidTaskIndexException();
+        }
+    }
+
+    /**
+     * Adds a new todo task to the TaskList.
+     *
+     * @param command The input command from the user.
+     * @return Formatted string containing the added todo task, to be displayed to the user.
+     * @throws InvalidFormatException        When the input command is of the wrong format.
+     * @throws EmptyTodoDescriptionException When the description is empty.
+     */
+    public String todo(String command) throws InvalidFormatException, EmptyTodoDescriptionException {
+        String[] matches = Parser.validateRegexAndMatch(command, "^todo (.+)", "todo {description}");
+        String description = matches[1].trim();
+        if (description.equals("")) {
+            throw new EmptyTodoDescriptionException();
+        }
+        Todo newTodo = new Todo(description);
+        this.tasks.add(newTodo);
+        return this.formatAddTaskString(newTodo);
+    }
+
+    /**
+     * Adds a new deadline task to the TaskList.
+     *
+     * @param command The input command from the user.
+     * @return Formatted string containing the added deadline task, to be displayed to the user.
+     * @throws InvalidFormatException When the input command is of the wrong format.
+     */
+    public String deadline(String command) throws InvalidFormatException {
```

Comment:
Perhaps a name that is a verb would be better here? To align with the coding conventions of names presenting methods being verbs.